File: backend/app/routes/inquiry.py
from fastapi import APIRouter, HTTPException
from app.supabase_client import supabase
from app.schemas.inquiry import (
    InquiryCreate,
    InquiryResponse,
)

router = APIRouter(
    prefix="/inquiry",
    tags=["Inquiry"],
)

@router.post("/", response_model=InquiryResponse)
def create_inquiry(
    inquiry: InquiryCreate,
):
    insert_response = (
        supabase.table("project_inquiries")
        .insert({
            "full_name": inquiry.full_name,
            "email": inquiry.email,
            "phone": inquiry.phone,
            "company": inquiry.company,
            "requirement": inquiry.requirement,
            "project_details": inquiry.project_details,
        })
        .execute()
    )

    if not insert_response.data:
        raise HTTPException(
            status_code=500,
            detail="Failed to create project inquiry."
        )

    return insert_response.data[0]

File: backend/src/routes/inquiry.ts
import { Router, Request, Response } from "express";
import { z } from "zod";
import { supabase } from "../supabase.js";

export const inquiryRouter = Router();

const inquirySchema = z.object({
  full_name: z.string(),
  email: z.string().email(),
  phone: z.string(),
  company: z.string(),
  requirement: z.string(),
  project_details: z.string(),
});

// POST /inquiry/
inquiryRouter.post("/", async (req: Request, res: Response): Promise<void> => {
  try {
    const validatedBody = inquirySchema.parse(req.body);

    const { data, error } = await supabase
      .from("project_inquiries")
      .insert({
        full_name: validatedBody.full_name,
        email: validatedBody.email,
        phone: validatedBody.phone,
        company: validatedBody.company,
        requirement: validatedBody.requirement,
        project_details: validatedBody.project_details,
      })
      .select();

    if (error || !data || data.length === 0) {
      res.status(500).json({ detail: error?.message || "Failed to create project inquiry." });
      return;
    }

    res.status(200).json(data[0]);
  } catch (error: any) {
    if (error instanceof z.ZodError) {
      res.status(422).json({ detail: error.errors });
      return;
    }
    res.status(500).json({ detail: error.message || "Internal server error" });
  }
});
